src/main/_docs/setup/setup-openshift.md

```diff
@@ -0,0 +1,99 @@
+---
+tags: [ "eclipse" , "che" ]
+title: OpenShift Installation
+excerpt: "Run Che on OpenShift."
+layout: docs
+permalink: /:categories/openshift/
+---
+{% include base.html %}
+
+You can run the Che server and its workspaces with OpenShift.
+OpenShift is built on top of Kubernetes and has built-in Docker registry, reverse proxy and OAuth server.
+OpenShift is ideal to deploy Che as enterprise wide cloud IDE.
+
+# Deploy Che on OpenShift Container Platform
+
+Use environment variables to set deployment options
+
+```shell
+export OPENSHIFT_ENDPOINT=<OCP_ENDPOINT_URL> # e.g. https://opnshmdnsy3t7twsh.centralus.cloudapp.azure.com:8443
+export OPENSHIFT_TOKEN=<OCP_TOKEN>
+export OPENSHIFT_NAMESPACE_URL=<CHE_HOSTNAME> # e.g. che-eclipse-che.52.173.199.80.xip.io
+export OPENSHIFT_FLAVOR=ocp
+```
+
+Download and run deployment scripts
+
+```shell
+DEPLOY_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/deploy_che.sh
+WAIT_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/wait_until_che_is_available.sh
+STACKS_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/replace_stacks.sh
+curl -fsSL ${DEPLOY_SCRIPT_URL} -o get-che.sh
+curl -fsSL ${WAIT_SCRIPT_URL} -o wait-che.sh
+curl -fsSL ${STACKS_SCRIPT_URL} -o stacks-che.sh
+bash get-che.sh && wait-che.sh && stacks-che.sh
+```
+
+# Deploy Che on Minishift
+
+Download and run deployment scripts
+
+```shell
+DEPLOY_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/deploy_che.sh
+WAIT_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/wait_until_che_is_available.sh
+STACKS_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/replace_stacks.sh
+curl -fsSL ${DEPLOY_SCRIPT_URL} -o get-che.sh
+curl -fsSL ${WAIT_SCRIPT_URL} -o wait-che.sh
+curl -fsSL ${STACKS_SCRIPT_URL} -o stacks-che.sh
+bash get-che.sh && wait-che.sh && stacks-che.sh
+```
+
+# Deploy Che on openshift.io
```

Comment:
Is this something we want users doing?

Reply:
That is how we (che-osio team) currently test before pushing to prod. But no we don't want osio user to do that. We want Che developers to be able to do that when testing a fix in OSIO. If you think that's confusing I can keep this part of the doc in github README https://github.com/redhat-developer/rh-che/blob/master/README.md#deploy-to-openshiftio

Reply:
Maybe just clarify that for developers testing changes to Che they should do this.

Reply:
Ok I've added a sentence to make it explicit.